lib/pbench/server/api/resources/metadata_api.py

```diff
@@ -0,0 +1,355 @@
+        {
+            "value": "blog text" <Client defined text, can be a string of json>,
+        }
+        Example: {"value": '{"config": "config", "description": "description"}'}
+
+        Authorization header can be included as
+            Authorization:   Bearer <Pbench authentication token (user received upon login)>
+        If the authorization header is not present, the created metadata session becomes public by default
+
+        :return: JSON Payload
+            response_object = {
+                    "message": "success"
+                    "data" {
+                        "id": "metadata_id",
+                        "value": "client text blob",
+                        "created": "datetime string",
+                        "updated": "datetime string",
+                        "key": favorite/saved
+                        }
+                }
+        """
+        metadata_key = key.upper()
+        if metadata_key not in [key.name for key in MetadataKeys]:
+            self.logger.warning(
+                "Invalid Metadata key provided during metadata creation. Key: {}",
+                metadata_key,
+            )
+            abort(
+                400,
+                message="Invalid metadata key. Key need to be either Favorite/Saved",
+            )
+
+        post_data = request.get_json()
+        if not post_data:
+            self.logger.warning("Invalid json object: {}", request.url)
+            abort(400, message="Invalid json object in request")
+
+        current_user = self.auth.token_auth.current_user()
+        if current_user:
+            current_user_id = current_user.id
+        else:
+            current_user_id = None
+
+        value = post_data.get("value")
+        if not value:
+            self.logger.warning(
+                "value not provided during metadata creation. user_id: {}",
+                current_user_id,
+            )
+            abort(400, message="Value field missing")
+
+        try:
+            # Create a new metadata session
+            metadata_session = Metadata(
+                value=value, key=metadata_key, user_id=current_user_id
+            )
+            # insert the metadata session for a user
+            metadata_session.add()
+            self.logger.info(
+                "New metadata session created for user_id {}", current_user_id
+            )
+        except Exception:
+            self.logger.exception("Exception occurred during the Metadata creation")
+            abort(500, message="INTERNAL ERROR")
+        else:
+            response_object = {
+                "message": "success",
+                "data": metadata_session.get_json(),
+            }
+            return make_response(jsonify(response_object), 201)
+
+    @Auth.token_auth.login_required(optional=True)
+    def get(self, key):
+        """
+        Get request for querying all the metadata sessions for a user.
```

Comment:
The terminology here is even murkier; you're returning all the values associated with a specified key. Above you used "sessions" (I think confusingly) for whatever metadata row you were creating, and following from that to here it would be natural to think of "metadata sessions for a user" as encompassing all metadata keys, while this is additionally constrained by the key name.

Reply:
Fixed